File: 523-continuous-subarray-sum/continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        // Cumulative sum array
        for(int i = 1; i < nums.size(); i++) {
            nums[i] += nums[i - 1];
        }

        // Modulus operation on cumulative sum array
        for(int i = 0; i < nums.size(); i++) {
            nums[i] %= k;
        }

        unordered_map<int, int> map;
        map[0] = -1; // To handle edge cases

        for(int i = 0; i < nums.size(); i++) {
            if(map.find(nums[i]) != map.end() && (i - map[nums[i]] > 1)) {
                return true;
            } else if(map.find(nums[i]) == map.end()) {
                map[nums[i]] = i;
            }
        }
        return false;
    }
};

File: 523-continuous-subarray-sum/continuous-subarray-sum.py
from typing import List

class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        # Cumulative sum array
        for i in range(1, len(nums)):
            nums[i] += nums[i - 1]
        
        # Modulus operation on cumulative sum array
        for i in range(len(nums)):
            nums[i] %= k
        
        map = {0: -1}  # To handle edge cases

        for i in range(len(nums)):
            if nums[i] in map and (i - map[nums[i]] > 1):
                return True
            elif nums[i] not in map:
                map[nums[i]] = i
        
        return False
